Security 1.0

aksoli666 opened the pull request:
added User entity and required dto`s
added method register and have repo, service, controller layers used data validation
added @FieldMatcher
added RegistrationException

aksoli666 committed:
Security 1.0
added User entity and required dto`s
added method register and have repo, service, controller layers
used data validation
added @FieldMatcher
added RegistrationException

Elena-Bruyako commented on src/main/java/shop/annatation/FieldMatch.java:
```diff
@@ -0,0 +1,23 @@
+package shop.annatation;
```
```suggestion
package shop.validation;
```

Elena-Bruyako commented on src/main/java/shop/validator/FieldMatchValidator.java:
```diff
@@ -0,0 +1,31 @@
+package shop.validator;
```
```suggestion
package shop.validation;
```

Elena-Bruyako commented on src/main/java/shop/validator/FieldMatchValidator.java:
```diff
@@ -0,0 +1,31 @@
+        String secondFieldValue = (String) new BeanWrapperImpl(field)
+                .getPropertyValue(secondFieldName);
+
+        if (firstFieldValue != null && secondFieldValue != null) {
```
You can use Objects.equals here

Elena-Bruyako commented on src/main/java/shop/dto/request/RegisterUserRequestDto.java:
```diff
@@ -0,0 +1,29 @@
+        secondFieldName = "repeatPassword",
+        message = "Password must match!"
+)
+public class RegisterUserRequestDto {
```
```suggestion
public class UserRegistrationRequestDto {
```

Elena-Bruyako commented on src/main/java/shop/dto/responce/RegisterUserResponseDto.java:
```diff
@@ -0,0 +1,21 @@
+
+@Setter
+@Getter
+public class RegisterUserResponseDto {
```
```suggestion
public class UserResponseDto {
```

Elena-Bruyako commented on src/main/java/shop/dto/responce/RegisterUserResponseDto.java:
```diff
@@ -0,0 +1,21 @@
+@Setter
+@Getter
+public class RegisterUserResponseDto {
+    @NotBlank
```
don't need validate response

Elena-Bruyako commented on src/main/java/shop/entity/User.java:
```diff
@@ -0,0 +1,39 @@
+    private String email;
+    @Column(nullable = false)
+    private String password;
+    @Column(
+            name = "first_name",
+            nullable = false
+    )
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/shop/entity/User.java:
```diff
@@ -0,0 +1,39 @@
+            nullable = false
+    )
+    private String firstName;
+    @Column(
+            name = "last_name",
+            nullable = false
+    )
```
```suggestion
    @Column(nullable = false)
```

Elena-Bruyako commented on src/main/java/shop/entity/User.java:
```diff
@@ -0,0 +1,39 @@
+            nullable = false
+    )
+    private String lastName;
+    @Column(name = "shipping_address")
```
```suggestion
```

Elena-Bruyako commented on src/main/java/shop/entity/User.java:
```diff
@@ -0,0 +1,39 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(
+            unique = true,
+            nullable = false
+    )
```
```suggestion
    @Column(unique = true, nullable = false)
```

Elena-Bruyako commented on src/main/java/shop/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public RegisterUserResponseDto register(RegisterUserRequestDto registerUserRequestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(registerUserRequestDto.getEmail()).isPresent()) {
```
```suggestion
        if (userRepository.existsByEmail(registerUserRequestDto.getEmail())) {
```

Elena-Bruyako commented on src/main/java/shop/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+            throw new RegistrationException("Can't register user with given email");
+        }
+        User user = userMapper.toUser(registerUserRequestDto);
+        User savedUser = userRepository.save(user);
+        return userMapper.toRegisterUserResponseDto(savedUser);
```
```suggestion
        userRepository.save(user);
        return userMapper.toRegisterUserResponseDto(user);
```

Elena-Bruyako commented on src/main/resources/db/changelog/changes/02.-create-users-table.yaml:
```diff
@@ -0,0 +1,39 @@
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
```
did you add 02 changelog to master changelog? 

Add empty line at the end of file

Elena-Bruyako reviewed (changes requested):
see comments

aksoli666 committed:
Security 2.0
added 02...yaml to master...yaml
refactor method in user repo
deleted response validation
rename all user dto`s
rename package with dto validation

ahoienko reviewed (approved):
Good job!
